### PR TITLE
nmp registry, open-source packages, and code security using npm@6

### PR DESCRIPTION
**Pull request by kimkimani**
nmp registry, open-source packages, and code security using npm@6

### CODE REVIEW

**Comment by kimkimani** [comment]
@hectorkambow submitted this PR but want to change submitted for PG #286 heading

**Comment by hectorkambow** [comment]
> @hectorkambow submitted this PR but want to change submitted for PG #286 heading

Sorry - Im misunderstanding - submit for PG?

**Comment by hectorkambow** [comment]
Also just finished running your article through our 3rd party plagiarism checker - and it raised a few red flags. 
Nothing major - but some sentences will need to be rephrased and or cited with proper sources! 
Some underlined sentences are obvious (negative red flags) but go over the pdf and see where it does apply and fix those accordingly. 
Thank you 
🚀 
[nmp registryopen-source packages, and code security using npm@6.pdf](https://github.com/section-io/engineering-education/files/5165501/nmp.registryopen-source.packages.and.code.security.using.npm%406.pdf)

**Comment by hectorkambow** [comment]
Plagiarism check DONE -with comments

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/npm-registry-opensource-code-security-with-npm6/hero.jpg
+    alt: Devtools example image Node.js debugger
+---
+Security is a huge concern for everyone. The safety of the code you use is important regardless of you being a beginner or expert in Node.js app development. It is even a professional habit that developers acquire as they grow with experience. Making sure open-source codes are secure is one of the fundamental mission of NPM, with approaches to security at scale to empower NPM users to develop secure code.
+<!--more-->
+
+With NPM(short for Node Package Manager), you do not need to worry about the safety of your code. NPM provides vulnerability-scanning tools that are built-in your Node.js workflow. These tools are faster and they automatically review every install request you make, it checks the Node Security Platform vulnerably database, and warns you if you try to use unsafe codes. It even analyzes your dependencies trees to check specifically what has security issues and recommend a replacement or fix the issues automatically.
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:20]: @kimkimani 
This sentence is a bit difficult to read - not sure what it was meant to say - please revisit and rephrase. 
"it checks the Node Security Platform vulnerably database"

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+In this article, we will have a look at NPM registry survey statistics, check other developers' responses to Node.js security practices and identify which tools you need to assess your open-source code safely.
+
+### NPM registry download statistics
+On September 29th, 2019, NPM marked 10 years of its existence. NPM has been used to develop every kind of application i.e. Web apps, mobile applications, Servers, Internet of things (IoT)/robotic, desktop applications and native mobile applications. This has greatly contributed to the NPM registry popularity. The number of libraries available in the NPM registry influences most developers to choose JavaScript.
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:25]: any sentences that have a date - please add a source 👍

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+![](/engineering-education/npm-registry-opensource-code-security-with-npm6/npm-influnce-statistics.png)
+
+[*Image Source*](https://medium.com/npm-inc/this-year-in-javascript-2018-in-review-and-npms-predictions-for-2019-3a3d7e5298ef#:~:text=npm%20has%20over%2010%20million,packages%20of%20open%20source%20JavaScript.)
+
+In June 2019, MPM crossed a million-libraries mark. With over 1.3 million libraries currently available, NPM registry is now the largest single collection of open source libraries in the world with more than 12 million developers.
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:30]: date needs a source (you can add as a link)

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+On April 23, 2018, NPM users downloaded five billion packages within the last 7 days of that date. On this day 4 years ago, that figure was 50 million, this means 10,000% growth. The average IP downloads were 2200 packages per month.
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:32]: source

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+![](/engineering-education/npm-registry-opensource-code-security-with-npm6/weekly-downloads.jpg)
+
+[*Image Source*](https://twitter.com/seldo/status/988477780441481217)
+
+Up to now, NPM has recorded tremendous growth with over 20 billion average weekly downloads. As am writing this article, the NPM registry recorded 88,820,279,735 libraries downloads last month. NPM registry is now serving around 125 billion requests at a whopping 6 Petabytes per month. [Ahmad Nassri](https://blog.npmjs.org/post/615388323067854848/so-long-and-thanks-for-all-the-packages).
+
+
+This statistical testaments, explain the resilience and commitment of NPM to keep the world's largest community safe by building a chain of fast reliable tools to help you perceive the safety of JavaScript code you write and share as well as the open-source code to ensure it is safe before you go live.
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md]: This paragraph is abit difficult to read through - please revisit and rephrase -

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+On April 23, 2018, NPM users downloaded five billion packages within the last 7 days of that date. On this day 4 years ago, that figure was 50 million, this means 10,000% growth. The average IP downloads were 2200 packages per month.
+![](/engineering-education/npm-registry-opensource-code-security-with-npm6/weekly-downloads.jpg)
+
+[*Image Source*](https://twitter.com/seldo/status/988477780441481217)
+
+Up to now, NPM has recorded tremendous growth with over 20 billion average weekly downloads. As am writing this article, the NPM registry recorded 88,820,279,735 libraries downloads last month. NPM registry is now serving around 125 billion requests at a whopping [6 Petabytes per month](https://blog.npmjs.org/post/615388323067854848/so-long-and-thanks-for-all-the-packages).
+
+This statistical testaments, explain the resilience and commitment of NPM to keep the world's largest community safe by building a chain of fast reliable tools to help you see the safety of JavaScript code you write and share as well as the open-source code to ensure it is safe before you go live.
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:39]: This paragraph was a bit difficult to get through - if you could rephrase.

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+## 2. Are you satisfied with the available methods to evaluate the safeness and the quality of open source code?
+
+It is interesting to note that more than half of JavaScript developers are not convinced with the existing tools for analyzing the security and quality of open-source code, which is very alarming.
+
+![](/engineering-education/npm-registry-opensource-code-security-with-npm6/oss-noncerned.png)
+
+[*Image Source*](https://medium.com/npm-inc/security-in-the-js-community-4bac032e553b)
+
+What about we separate the response dependent on the concern of open source code or their own code (the code you write).
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:74]: Please revisit this sentence:
"What about we separate the response dependent on the concern of open source code or their own code (the code you write)."

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+If all answers are combined, a measure of how many developers uses some form of security practices on codes depending on experience will be:
+![](/engineering-education/npm-registry-opensource-code-security-with-npm6/level-of-experience.png)
+
+[*Image Source*](https://medium.com/npm-inc/security-in-the-js-community-4bac032e553b)
+
+### Conclusion
+-   The best code security practices are common to experienced developers. These practices include code review, linting and testing.
+-  Practices such as testing, linting, code surveys, and security examines are indicators  of a polished developer.
+-   Solid greater part of this developers are more likely concerned with the safety of the code they write as well as the open-source code they use.
```
Comment (hectorkambow) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:109]: Please revisit and rephrase this sentence: 
"Solid greater part of this developers are more likely concerned with the safety of the code they write as well as the open-source code they use."

**Comment by hectorkambow** [comment]
Very solid article -great start - there were a few grammatical errors and I changed some other minor things for readability. 
If you could go over my comments - update those sentences - and give the article another revision before we pass to a tech review. 

We want to make sure the audience is following your thoughts as you intended and are not left confused - about what you meant. 

Also I believe there are Two conclusions - please double check and change. 🚀 

Let me know if you have any questions 👍

**Comment by kimkimani** [comment]
@hectorkambow Thank you for the edits, they have improved the reading flow.
I have restructured the difficult to read sentences/paragraphs and added date references.

**Comment by hectorkambow** [comment]
Good morning (afternoon) @jstangroome  - got an NMP security article - I wanted to ask if you could give a tech review whenever you have a moment next week (before we publish) 
Thank you 🚀 
Hope you have a great weekend! 👋

File: articles/npm-registry-opensource-code-security-with-npm6/index.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+-   You can do a one-off, temporary installs of command-line utilities that are rarely ran, [such as generators](https://blog.npmjs.org/post/173240511455/the-new-npm-cli-a-year-in-review-or-what-you):
+
+ `$ npx create-react-app`.
+
+-   You can easily try different versions of these tools with a single install if you need to [compare them](https://blog.npmjs.org/post/173240511455/the-new-npm-cli-a-year-in-review-or-what-you): 
+
+`$ npx standard@8 && npx standard@10`.
+
+-   You can even install the [shell auto-fallback](https://t.umblr.com/redirect?z=https%3A%2F%2Fwww.npmjs.com%2Fpackage%2Fnpx%23shell-auto-fallback&t=MTVkODRiYmJkOWQ0ZTQxMjYwZGU3M2U0OGFiMDAwZTQ3YjAzMjc5YyxSQlc3OFpFUg%3D%3D&b=t%3AnXsLs1P4AptPf1fBr_nFxw&p=https%3A%2F%2Fblog.npmjs.org%2Fpost%2F173240511455%2Fthe-new-npm-cli-a-year-in-review-or-what-you&m=1&ts=1598809717) to not even have to write `$ npx...` in many cases.
```
Comment (jstangroome) [inline articles/npm-registry-opensource-code-security-with-npm6/index.md:168]:
```suggestion
-   You can even install the [shell auto-fallback](https://www.npmjs.com/package/npx#shell-auto-fallback) to not even have to write `$ npx...` in many cases.
```

**Review by jstangroome — APPROVED** [review]

**Pull request merged**